user/tests/integration/wiring/no_fixture_i2c/i2c.cpp

```diff
@@ -0,0 +1,131 @@
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+//SerialLogHandler g_logSerial(LOG_LEVEL_ALL);
+
+static const int WIRE_ACQUIRE_BUFFER_SIZE = HAL_PLATFORM_I2C_BUFFER_SIZE(HAL_I2C_INTERFACE1) + 1;
+static const int WIRE1_ACQUIRE_BUFFER_SIZE = HAL_PLATFORM_I2C_BUFFER_SIZE(HAL_I2C_INTERFACE2) + 1;
+static const int WIRE1_TRACKER_MINIMUM_BUFFER_SIZE = 512;
+
+static hal_i2c_config_t allocateWireConfig(uint32_t bufferSize) {
+    hal_i2c_config_t config = {
+        .size = sizeof(hal_i2c_config_t),
+        .version = HAL_I2C_CONFIG_VERSION_2,
+        .rx_buffer = new (std::nothrow) uint8_t[bufferSize],
+        .rx_buffer_size = bufferSize,
+        .tx_buffer = new (std::nothrow) uint8_t[bufferSize],
+        .tx_buffer_size = bufferSize,
+        .flags = HAL_I2C_CONFIG_FLAG_FREEABLE
+    };
+    return config;
+}
+
+static void freeWireBuffers(hal_i2c_config_t * config){
+    free(config->tx_buffer);
+    free(config->rx_buffer);
+}
+
+hal_i2c_config_t acquireWireBuffer()
+{
+    return allocateWireConfig(WIRE_ACQUIRE_BUFFER_SIZE);
+}
+
+#if Wiring_Wire1
+hal_i2c_config_t acquireWire1Buffer()
+{
+    return allocateWireConfig(WIRE1_ACQUIRE_BUFFER_SIZE);
+}
+#endif
+
+test(I2C_00_hal_init_default_buffer)
+{
+    // Initializing the HAL with a null config is allowed and should allocate system defaults
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, nullptr), (int)SYSTEM_ERROR_NONE);
+
+#if (PLATFORM_ID == PLATFORM_TRACKER) || (PLATFORM_ID == PLATFORM_TRACKERM)
+    // Tracker platforms should have buffers at least 512 bytes large, allocated by the system on startup
+    // IE trying to re-initialize the HAL with same size should fail
+    hal_i2c_config_t wire1_small_buffers = allocateWireConfig(WIRE1_TRACKER_MINIMUM_BUFFER_SIZE);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE2, &wire1_small_buffers), (int)SYSTEM_ERROR_NOT_ENOUGH_DATA);
```

Comment:
This is not going to work on Tracker M, there is only 1 I2C peripheral. You should use `HAL_PLATFORM_EXTERNAL_RTC_I2C` if that's what is being used for tests on these platforms.

Reply:
My understanding of the requirements from the original ticket [sc-109914](https://app.shortcut.com/particle/story/109914/allow-user-app-to-override-i2c-buffer-on-tracker-trackerm-other-platforms-that-may-have-rtc-and-pmic) is that the i2c busses that connect to peripherals that return >32 bytes of data should have 512 byte buffers by default. 
The only device that returns >32 bytes of data currently is the new Quectel GNSS chip.
This chip is only present on Trackerm with its single I2C interface, and I think it will be on a new tracker variant on I2C2. So this test should be testing those two specific instances.